projects/vanilla/src/app/configurators/metadata-configurator.component.ts

```diff
@@ -7,43 +7,132 @@ import { ComponentConfig, ConfiguratorContext } from "@sinequa/ngx-ui-builder";
   selector: 'sq-metadata-configurator',
   template: `
 <div class="form-group">
-  <label for="item">Item</label>
-  <select id="item" class="form-select mb-2" [(ngModel)]="config.item" (ngModelChange)="configChanged(true)">
-    <option *ngFor="let a of metadata" [ngValue]="a.field">{{a.field}}</option>
-  </select>
-
-  <uib-checkbox [context]="context" property="clickable" label="Clickable"></uib-checkbox>
-  <uib-checkbox [context]="context" property="showTitle" label="Show title"></uib-checkbox>
-  <uib-checkbox [context]="context" property="showIcon" label="Show icon"></uib-checkbox>
-
-  <div *ngIf="isEntity">
-    Entity options:
-    <uib-checkbox [context]="context" property="showCounts" label="Show counts"></uib-checkbox>
-    <uib-checkbox [context]="context" property="showEntityTooltip" label="Show tooltip"></uib-checkbox>
+  <label class="form-label mt-2">Selection</label>
+  <p class="small text-muted m-0">Select which metadata to display. The order can be changed by dragging the following options up and down.</p>
+  <uib-multi-selector [options]="metadataSelection" [enableReorder]="true" [(ngModel)]="config.items" (ngModelChange)="updatedList()">
+  </uib-multi-selector>
+
+  <div *ngIf="config.items?.length">
+    <div class="input-group mb-2">
+      <input type="text" class="form-control" [(ngModel)]="metadataString" placeholder="Type a string to insert between metadatas">
+      <div class="input-group-append">
+        <button class="btn btn-secondary" (click)="insertString()">Insert a string</button>
+      </div>
+    </div>
```

Comment:
I'm not sure about this feature, because it makes the component hard to understand (I didn't understand what the "Insert string" button when looking at the screenshot, until I looked at the code).
Also, if this component lets you insert a string, it should let you edit or delete the string.

Reply:
The string gets removed automatically if you untoggle it from the ordering list. But I can indeed allow its edition yes, so far we don't see the simple strings from the metadata to edit list, I could make an input to edit it when we select one.

As for the understandability, I'm not sure? From your comment it seems you think we should maybe remove it, but it can be handy when you know it, maybe it's mostly a matter of presenting it better

Reply:
The thing is that it does feel very natural to configure the metadata string like that. I would expect something like a text box where you could write the text with placeholders for the metadata. Something like:

```
Document {{filename}} modified on {{modified}} by {{author}}
```

But even this is dangerous because it doesn't handle when `filename`, `modified` or `author` are undefined